core/src/main/java/io/questdb/cairo/ColumnPurgeJob.java

```diff
@@ -251,12 +268,16 @@ private void putTasksFromTableToQueue() {
                         taskRun.appendColumnInfo(columnVersion, partitionTs, partitionNameTxn, rec.getUpdateRowId());
                     }
                     if (taskRun != null) {
-                        retryQueue.add(taskRun);
+                        columnPurgeOperator.purgeExclusive(taskRun);
+                        taskPool.push(taskRun);
                     }
                 }
+
+                if (count > 0) {
+                    LOG.info().$("cleaned up rewritten column files [cleanCount=").$(count).I$();
+                }
             }
-            if (retryQueue.size() == 0 && writer != null) {
-                // No tasks to do. Cleanup the log table
+            if (writer != null) {
                 try {
                     writer.truncate();
```

Comment:
Looks like incomplete entries older than `PurgeRetryLimitDays` are ignored and possibly lost as the table is unconditionally truncated on every restart.

Reply:
I think it's fine. If we couldn't delete file for a month, what are the chances to delete it after that.

Reply:
remove the limit